metadata-ingestion/src/datahub/ingestion/source/sql/mssql/job_models.py

```diff
@@ -11,18 +15,65 @@
 
 @dataclass
 class ProcedureDependency:
+    flow_id: str
+    env: str
+    server: Optional[str]
     db: str
     schema: str
     name: str
     type: str
-    env: str
-    server: Optional[str]
+    incoming: Union[str, int]
+    outgoing: Union[str, int]
     source: str = "mssql"
 
+    def __post_init__(self):
+        self.incoming = int(self.incoming)
+        self.outgoing = int(self.outgoing)
+
 
 @dataclass
 class ProcedureLineageStream:
-    dependencies: List[ProcedureDependency]
+    dependencies: List[ProcedureDependency] = field(default_factory=list)
+
+    @property
+    def as_input_datasets(self) -> List[str]:
+        return [
+            make_dataset_urn_with_platform_instance(
+                platform=dep.source,
```

Comment:
dep.source is always mssql right?

Reply:
Correct. Because procedure dependencies are extraqcting by _populate_stored_procedures_dependencies method, which query is built around two mssql objects:
- [sys.procedures](https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-procedures-transact-sql?view=sql-server-ver16)
- [sys.dm_sql_referenced_entities](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-sql-referenced-entities-transact-sql?view=sql-server-ver16)
- [sys.objects](https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-objects-transact-sql?view=sql-server-ver16)
So, we assume, that all related entities are mssql objects by default.